### ISSUE TITLE
Update to meshkernelpy 5.0.1

### ISSUE THREAD

**Issue by veenstrajelmer** [Deltares/HYDROLIB-core#702]
**What is the need for this task.**
Major version of meshkernelpy was updated (to 5.0.0), hydrolib-core should be accomodated for this new version. It is the expectation that no code changes are needed.

**What is the task?**
Update minimum meshkernelpy version in pyproject.toml

Also, add testcase for `DeleteMeshOption.FACES_WITH_INCLUDED_CIRCUMCENTERS`

**Comment by veenstrajelmer** [comment]
Two testcases fail:
```
FAILED tests/dflowfm/test_net.py::test_create_clip_2d[0-False-22-68-13] - assert 30 == 22
FAILED tests/dflowfm/test_net.py::test_create_clip_2d[1-False-30-98-20] - assert 22 == 30
```

**Comment by veenstrajelmer** [comment]
This is resolved in meshkernel 5.0.1, released today
